src/fio/FioEngine.ts

```diff
@@ -235,24 +235,20 @@ export class FioEngine extends CurrencyEngine<FioTools, SafeFioWalletInfo> {
         privateKeys: JsonObject,
         ...rest: any[]
       ): Promise<ObtData[]> => {
-        const obtDecoder = new GetObtData({
-          fioPublicKey: this.walletInfo.keys.publicKey,
-          getEncryptKey: this.defaultGetEncryptKey,
-          includeEncrypted: false
-        })
-        const encryptedObtData = await this.fetchEncryptedObtData(
-          'getObtData',
-          obtDecoder
+        const fioSdk = new FIOSDK(
+          privateKeys.fioKey,
+          this.walletInfo.keys.publicKey,
+          this.networkInfo.historyNodeUrls,
+          this.fetchCors
         )
-        obtDecoder.privateKey = privateKeys.fioKey
-        obtDecoder.publicKey = this.walletInfo.keys.publicKey
-        const decryptedObtData: { obt_data_records: ObtData[] } =
-          (await obtDecoder.decrypt({
-            obt_data_records: encryptedObtData,
-            more: 0
-          })) ?? { obt_data_records: [] }
-
-        return decryptedObtData.obt_data_records
+
+        const res = await fioSdk.genericAction('getObtData', {
+          limit: 100,
+          offset: 0,
```

Comment:
This implementation will only query 100 records and stop. Is that appropriate here? If you don't ultimately need fetchEncryptedObtData in that case you can remove it.

Also, while not really important here, calling fioSdk.getObtData() directly instead of genericAction gives you return types instead of any.